rmgpy/solver/surface.pyx

```diff
@@ -384,17 +409,32 @@ cdef class SurfaceReactor(ReactionSystem):
 
         C = np.zeros_like(self.core_species_concentrations)
         V = self.V  # constant volume reactor
+        A = self.V * surface_volume_ratio_si  # area
+        total_sites = self.surface_site_density.value_si * A  # todo: double check units
 
+        current_surface_coverages = {}
         for j in range(num_core_species):
             if species_on_surface[j]:
                 C[j] = (N[j] / V) / surface_volume_ratio_si
+                surface_site_fraction = N[j] / total_sites
+                species = cov_dep_index_species[j]
+                current_surface_coverages[species] = surface_site_fraction
```

Comment:
I think we should try to avoid all this dictionary making/filling/reading inside the residual function of the solver (which we want to be very fast). I'll sleep on it.

Reply:
No, I disagree